packages/zoe/src/contractSupport/atomicTransfer.js

```diff
@@ -0,0 +1,132 @@
+const { Fail, quote: q } = assert;
+
+/**
+ * @typedef {[
+ *   fromSeat?: ZCFSeat,
+ *   toSeat?: ZCFSeat,
+ *   fromAmounts?: AmountKeywordRecord,
+ *   toAmounts?: AmountKeywordRecord
+ * ]} TransferArgs
+ */
+
+/**
+ * TODO Refactor from being a helper into being zcf's replacement for
+ * reallocate. Is currently a helper during the transition, to avoid
+ * interference with progress on Zoe durability.
+ *
+ * See the helpers below, `fromOnly`, `toOnly`, and `atomicTransfer`,
+ * which will remain helpers, for further conveniences for expressing
+ * atomic rearragements clearly.
+ *
+ * @param {ZCF} zcf
+ * @param {TransferArgs[]} transfers
+ */
+export const atomicRearrange = (zcf, transfers) => {
+  const uniqueSeatSet = new Set();
+  for (const [
+    fromSeat = undefined,
+    toSeat = undefined,
+    _subtrahend,
+    _addend,
```

Comment:
If you're going to use `fromAmounts` and `toAmounts` below, this should too.

Reply:
Done